ui/packages/ui/src/app/shared/Utils.ts

```diff
@@ -29,6 +36,74 @@ export function getConnectorTypeDescription(connType: ConnectorType): string {
   return "Unknown type";
 }
 
+/**
+ * Get a new ConnectorConfiguration for the specified ConnectorType
+ * @param connType the connector type
+ */
+export function newConnectorConfiguration(connectorType: ConnectorType): ConnectorConfiguration {
+  let connectorConfig = null;
+  if (connectorType) {
+    connectorConfig = { 
+      "name": "tempName",
+      "config": {
+        "connector.class": connectorType.className
+      }
+    } as ConnectorConfiguration;
+  }
+  return connectorConfig;
+}
+
+/**
+ * Get property definitions for the supplied category
+ * @param propertyDefns the array of all ConnectorProperty objects
+ * @param category the category for narrowing the ConnectorProperty objects
+ */
+export function getPropertyDefinitions(propertyDefns: ConnectorProperty[], category: PropertyCategory): ConnectorProperty[] {
+  const connProperties: ConnectorProperty[] = [];
+  for (const propDefn of propertyDefns) {
+    if (isInCategory(propDefn, category)) {
+      connProperties.push(propDefn);
+    }
+  }
+  return connProperties;
+}
+
+export function isInCategory(propertyDefn: ConnectorProperty, category: PropertyCategory) {
```

Comment:
Hoping that this logic moves to the backend. If not we could put this in a separate file of itself(as down the line more prop may also be added). WDYT?

Reply:
@indraraj yes if we are ultimately forced to leave this on ui side, it should be separated out better.  My thinking is that this is just a temporary solution so that we can keep making progress on the components.  I'll just go ahead and merge this - maybe I will redo a little today if time permits